packages/net/udp_socket.pony

```diff
@@ -406,3 +406,116 @@ actor UDPSocket
       @pony_os_socket_close[None](_fd)
       _fd = -1
     end
+
+  fun getsockopt(level: I32, option_name: I32, option: Array[U8]): (U32, U32) =>
+    """
+    General wrapper for UDP sockets to the `getsockopt(2)` system call.
+
+    In case of system call success, this function returns the 2-tuple:
+    1. The integer `0`.
+    2. The value of the `*(uint32_t)option_length` argument set by
+       `getsockopt(2)`.  The caller must use this length to properly
+        interpret the bytes written by the kernel into the `option`
+        byte array: this length may be smaller than `option`'s
+        original size.
+
+    In case of system call failure, this function returns the 2-tuple:
+    1. The value of `errno`.
+    2. An undefined value that must be ignored.
+    """
+    _OSSocket.getsockopt(_fd, level, option_name, option)
+
+  fun getsockopt_u32(level: I32, option_name: I32): (U32, U32) =>
+    """
+    Wrapper for UDP sockets to the `getsockopt(2)` system call where
+    the kernel's returned option value is a C `uint32_t` type / Pony
+    type `U32`.
+
+    In case of system call success, this function returns the 2-tuple:
+    1. The integer `0`.
+    2. The `*option_value` returned by the kernel converted to a Pony `U32`.
+
+    In case of system call failure, this function returns the 2-tuple:
+    1. The value of `errno`.
+    2. An undefined value that must be ignored.
+    """
+    _OSSocket.getsockopt_u32(_fd, level, option_name)
+
+  fun setsockopt(level: I32, option_name: I32, option: Array[U8]): U32 =>
```

Comment:
should all those methods setting sock opts be `fun ref` instead of `fun box` as they kind of mutate the connection (although it is an actor)?

Reply:
Oops, yes.  And that was also mentioned back in a Sync meeting, sorry, I'll fix it up shortly.